_posts/2021-05-26-introduction-salesforce-commerce-api.markdown

```diff
@@ -0,0 +1,346 @@
+
+
+# Quelques appels
+
+Nous voici dans la partie intéressante : nous avons les informations de connexion et nous pouvons nous amuser avec les API classiques. Je ne vais pas présenter ici toutes les API mais une ou deux afin de se faire la main. Pour le reste je vous laisse consulter la documentation officielle.
+
+Le but sera de rechercher un produit et de l’ajouter au panier. En affichant les API de type “Shopper” dans la documentation officielle on voit 10 résultats. Dans ces résultats on trouve ce qui nous intéresse : “Shopper Search” et “Shopper Basket” (on ne va pas aller jusqu’au passage de commande, mais si vous voulez aller jusque là vous avez aussi “Shopper Orders”).
+
+Avec Shopper Search nous allons rechercher un produit dans le catalogue puis nous allons créer un panier et ajouter un des produits dans ce panier avec Shopper Basket.
+
+## Shopper Search
+
+Tout commence par l’onglet [API Specification de la documentation][shopper_search_documentation]. Celui-ci donne l’URL de base de l’API (toutes les API ont une URL spécifique), ici : ```https://{shortCode}.api.commercecloud.salesforce.com/search/shopper-search/{version}```
+
+Si vous sélectionnez les endpoints disponibles dans le menu vous trouverez “GET productSearch” donnant les détails de ce qu’il faut envoyer et à quelle URL :
+- Type : ```GET```
+- URL : ```/organizations/{organizationId}/product-search```
+- Header requis : le token Bearer
+- Paramètres :
+  - ```siteId``` : requis (l’identifiant du site Salesforce à requêter)
+  - ```q``` : non requis mais on va le fournir (il s’agit de la recherche à faire)
+
+Il suffit donc tout simplement de s’authentifier (guest ou non) puis d’exécuter cette requête avec le token reçu et la recherche que vous souhaitez faire (paramètre “q”). L’API vous retournera un JSON avec le résultat de la recherche ! Pour ma part je vais prendre le premier produit retourné et l’ajouter au panier.
+
+## Shopper Basket
+
+La création d'un panier (basket en anglais) peut se faire en une ou plusieurs fois :
+- soit vous avez toutes les informations à mettre dans le panier (produits, modes de livraison, etc) et vous pouvez tout envoyer en un appel
+- soit vous créez un panier vide et le modifiez au fur et à mesure
+
+Dans notre cas nous allons choisir l'option 2. Nous allons donc faire un premier appel pour créer le panier. Comme dans la partie précédente tout commence par la [documentation Shopper Basket][documentation_shopper_basket]. Dans l'onglet Developer Guide vous trouverez beaucoup d'informations intéressentes sur comment procéder, je vous conseille de le lire si vous êtes pas au point sur cette notion de panier et de tout ce qu'il est censé contenir. Pour les autres, rendez-vous sur l'onglet "API Specification" pour trouver l'URL de base des API Basket: `https://{shortCode}.api.commercecloud.salesforce.com/checkout/shopper-baskets/{version}
```

Comment:
It seems that there is a missing "`" at the end of the line to close the code formatting